nodes/generator/torusKnot.py

```diff
@@ -0,0 +1,428 @@
+# ##### BEGIN GPL LICENSE BLOCK #####
+#
+#  This program is free software; you can redistribute it and/or
+#  modify it under the terms of the GNU General Public License
+#  as published by the Free Software Foundation; either version 2
+#  of the License, or (at your option) any later version.
+#
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License
+#  along with this program; if not, write to the Free Software Foundation,
+#  Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+#
+# ##### END GPL LICENSE BLOCK #####
+
+# This node is based on blender's built-in TorusKnot+ add-on
+
+import bpy
+from bpy.props import IntProperty, FloatProperty, BoolProperty, EnumProperty
+
+from math import sin, cos, pi, sqrt, radians
+from mathutils import *
+from random import random
+import time
+
+from sverchok.node_tree import SverchCustomTreeNode
+from sverchok.data_structure import updateNode, match_long_repeat, SvSetSocketAnyType
+
+DEBUG=False
+
+options_plus=True
+
+# greatest common denominator
+def gcd(a, b):
```

Comment:
Ha.. i didn't know python math has a gcd. Actually it seems that the 2.7
does not have it. I can remove it and use the math version.